pkg/telemetry/mocks/mock_tracing_server.go

```diff
@@ -0,0 +1,48 @@
+package mocks
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net"
+	"sync"
+
+	otlpcollector "go.opentelemetry.io/proto/otlp/collector/trace/v1"
+	"google.golang.org/grpc"
+)
+
+type MockTracingServer struct {
```

Comment:
```suggestion
type mockTracingServer struct {
```

Reply:
You know what, why not just generate a mock for this using mockgen and assert that the `Export` function is called exactly once? I think that'd be more preferable than writing custom mock code and maintaining the mocked code.

I recommend we add a `go:generate mockgen ...` directive at the top of the test file that requires this mocked interface, and export the mocked interface to the `internal/mocks` package.

```
go:generate mockgen -destination <path>/mock_trace_service_server.go -package mocks  go.opentelemetry.io/proto/otlp/collector/trace/v1 TraceServiceServer
```

Then we can do something like:
```go
mockController := gomock.NewController(t)
defer mockController.Finish()

mockTraceServiceServer := mockstorage.NewMockTraceServiceServer(mockController)

mockTraceServiceServer.EXPECT().Export(...).Times(1)
```

Reply:
This is a good idea, but the mock has to listen on a specific port or else the thing fails: https://github.com/openfga/openfga/blob/df6f05a3d2ac1d8472079057d4b73c3e337f59ce/pkg/telemetry/tracing.go#L68-L75

I tried having the mock listen on a port with

```go
func TestBuildServiceWithTracingEnabled(t *testing.T) {
	// create mock OTLP server
	mockController := gomock.NewController(t)
	defer mockController.Finish()
	mockTraceServiceServer := mocks2.NewMockTraceServiceServer(mockController)
	mockTraceServiceServer.EXPECT().Export(gomock.Any(), gomock.Any()).Times(1)

	// attach it to a port
	otlpServerPort, otlpServerPortReleaser := TCPRandomPort()
	localOTLPServerURL := fmt.Sprintf("localhost:%d", otlpServerPort)
	otlpServerPortReleaser()

	go func() {
		lis, err := net.Listen("tcp", fmt.Sprintf(":%d", otlpServerPort))
		if err != nil {
			log.Fatalf("failed to listen: %v", err)
		}
		server := grpc.NewServer()
		otlpcollector.RegisterTraceServiceServer(server, mockTraceServiceServer) // <-------- fails here
		if err := server.Serve(lis); err != nil {
			log.Fatalf("failed to serve: %v", err)
		}
	}()

	// create OpenFGA server with tracing enabled
	cfg := MustDefaultConfigWithRandomPorts()
	cfg.Trace.Enabled = true
	cfg.Trace.SampleRatio = 1
	cfg.Trace.OTLP.Endpoint = localOTLPServerURL
```

However, the code doesn't build. 

```
cmd/run/run_test.go:405:52: cannot use mockTraceServiceServer (variable of type *"github.com/openfga/openfga/internal/mocks".MockTraceServiceServer) as "go.opentelemetry.io/proto/otlp/collector/trace/v1".TraceServiceServer value in argument to otlpcollector.RegisterTraceServiceServer: 

*"github.com/openfga/openfga/internal/mocks".MockTraceServiceServer does not implement "go.opentelemetry.io/proto/otlp/collector/trace/v1".TraceServiceServer (missing method mustEmbedUnimplementedTraceServiceServer)
```

i'm not sure why it complains because the mock does have an implementation of `mustEmbedUnimplementedTraceServiceServer`

Reply:
Taking a look 👀

Reply:
https://github.com/uber/mock/issues/23